ios/app/brave_core_main.mm

```diff
@@ -95,9 +109,30 @@ - (instancetype)initWithUserAgent:(NSString*)userAgent
     web::SetWebClient(_webClient.get());
 
     _delegate.reset(new BraveMainDelegate());
-    _delegate->SetSyncServiceURL(base::SysNSStringToUTF8(syncServiceURL));
 
     web::WebMainParams params(_delegate.get());
+    NSMutableArray* arguments =
+        [[[NSProcessInfo processInfo] arguments] mutableCopy];
+    NSMutableArray* switches = [[NSMutableArray alloc] init];
+    for (NSString* key in additionalSwitches) {
+      if (![additionalSwitches[key] isKindOfClass:NSString.class]) {
```

Comment:
This should do a `DCHECK` right before the `continue` statement because this should be invalid behaviour.